test/misc.jl

```diff
@@ -16,91 +16,103 @@ function mysqrt(x::AbstractArray)
     return sqrt.(x)
 end
 
-forward(x) = mysqrt(x), 0
-conditions(x, y, z) = y .^ 2 .- x
-implicit = ImplicitFunction(forward, conditions)
-
-# Skipped because of https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/232 and because it detects weird type instabilities
-@testset verbose = true "ChainRulesTestUtils.jl" begin
-    @test_skip test_rrule(implicit, x)
-    @test_skip test_rrule(implicit, X)
+function get_implicit_func(::Val{byproduct}) where {byproduct}
+    if byproduct
+        forward = (x) -> (mysqrt(x), 0)
+        conditions = (x, y, z) -> y .^ 2 .- x
+        implicit = ImplicitFunction(forward, conditions, Val(true))
+    else
+        forward = (x) -> mysqrt(x)
+        conditions = (x, y) -> y .^ 2 .- x
+        implicit = ImplicitFunction(forward, conditions)
+    end
+    return implicit
 end
 
-@testset verbose = true "Vectors" begin
-    x = rand(2)
-    y, _ = implicit(x)
-    J = Diagonal(0.5 ./ sqrt.(x))
+@testset "byproduct = $byproduct" for byproduct in (Val(true), Val(false))
+    implicit = get_implicit_func(byproduct)
+    # Skipped because of https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/232 and because it detects weird type instabilities
```

Comment:
Any idea what to do about these?

Reply:
I need to look into it. Perhaps in another PR.